 
<launch>

  <arg name="robot_model"                       default=""/>
  <arg name="robot_name"                        default="$(arg robot_model)"/>
  <arg name="base_link_frame"                   default="base_link"/>
  <arg name="use_world_frame"                   default="true"/>
  <arg name="external_urdf_loc"                 default=""/>
  <arg name="use_joint_pub"                     default="false"/>
  <arg name="use_joint_pub_gui"                 default="false"/>
  <arg name="use_rviz"                          default="true"/>
  <arg name="load_gazebo_configs"               default="false"/>
  
  <arg name="env_runner" value="$(eval 'env_run' if not optenv('OS', 'unknown').lower().startswith('windows') else 'env_run.bat')" />
  <!-- the following seems to work too when in devel space, but not in install_isolated -->
  <!-- <arg name="env_runner" value="env_run" /> -->
  <arg name="config" default="front_laser" />
  
  
  <arg name="rvizconfig"                        default="$(find interbotix_xsturret_descriptions)/rviz/xsturret_description.rviz" />
  <arg name="model"                             default="$(find pckg)/urdf/$(arg robot_model).xacro
                                                         robot_name:=$(arg robot_name)
                                                         base_link_frame:=$(arg base_link_frame)
                                                         use_world_frame:=$(arg use_world_frame)
                                                         external_urdf_loc:=$(arg external_urdf_loc)
                                                         load_gazebo_configs:=$(arg load_gazebo_configs)"/>
                                                         
                                                         

  <param name="$(arg robot_name)/robot_description" command="$(find jackal_description)/scripts/$(arg env_runner)
                    $(find jackal_description)/urdf/configs/$(arg config)
                    $(find xacro)/xacro $(arg model)
                    --inorder"/>

  <node if="$(arg use_joint_pub)"
    name="joint_state_publisher"
    pkg="joint_state_publisher"
    type="joint_state_publisher"
    ns="$(arg robot_name)">
  </node>

  <node if="$(arg use_joint_pub_gui)"
    name="joint_state_publisher_gui"
    pkg="joint_state_publisher_gui"
    type="joint_state_publisher_gui"
    ns="$(arg robot_name)">
  </node>

  <node
    name="robot_state_publisher"
    pkg="robot_state_publisher"
    type="robot_state_publisher"
    ns="$(arg robot_name)">
  </node>

  <node if="$(arg use_rviz)"
    name="rviz"
    pkg="rviz"
    type="rviz"
    args="-d $(arg rvizconfig)"
    ns="$(arg robot_name)"/>

</launch>
